src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,501 @@
+                                               WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+    }
+}
+
+uint16_t WebRTCTransportProviderServer::GenerateSessionID()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = lastSessionId++;
+
+    // Handle wrap-around per spec
+    if (lastSessionId > kMaxSessionId)
+    {
+        lastSessionId = 0;
+    }
+
+    // Ensure uniqueness
+    while (FindSession(candidateId) != nullptr)
+    {
+        candidateId = lastSessionId++;
+        if (lastSessionId > kMaxSessionId)
+        {
+            lastSessionId = 0;
+        }
+    }
+
+    return candidateId;
+}
+
+// Command Handlers
+void WebRTCTransportProviderServer::HandleSolicitOffer(HandlerContext & ctx, const Commands::SolicitOffer::DecodableType & req)
+{
+    // According to the Matter WebRTC Transport Provider specification,
+    // at least one of VideoStreamID or AudioStreamID must be present.
+    if (!req.videoStreamID.HasValue() && !req.audioStreamID.HasValue())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Prepare arguments for Delegate::HandleSolicitOffer using OfferRequestArgs struct
+    Delegate::OfferRequestArgs args;
+    args.id                 = GenerateSessionID();
+    args.streamUsage        = req.streamUsage;
+    args.videoStreamID      = req.videoStreamID;
+    args.audioStreamID      = req.audioStreamID;
+    args.iceServers         = req.ICEServers;
+    args.iceTransportPolicy = req.ICETransportPolicy;
+    args.metadataOptions    = req.metadataOptions;
+    args.peerNodeID         = GetNodeIdFromCtx(ctx.mCommandHandler);
+    args.peerFabricIndex    = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    // Delegate processing
+    WebRTCSessionTypeStruct outSession;
+    bool deferredOffer = false;
+
+    auto status = Protocols::InteractionModel::ClusterStatusCode(mDelegate->HandleSolicitOffer(args, outSession, deferredOffer));
+    if (!status.IsSuccess())
+    {
+        // Delegate encountered a problem (out of resources, invalid data, etc.)
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Store or update the WebRTCSessionTypeStruct in the CurrentSessions
+    if (UpsertSession(outSession) == UpsertResultEnum::kInserted)
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Inserted a new session with ID=%u.", outSession.id);
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Updated existing session with ID=%u.", outSession.id);
+    }
+
+    // Send the response
+    Commands::SolicitOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.deferredOffer   = deferredOffer;
+    resp.videoStreamID.SetValue(outSession.videoStreamID);
+    resp.audioStreamID.SetValue(outSession.audioStreamID);
```

Comment:
This is a spec violation.  Please check the conformance on these fields carefully.